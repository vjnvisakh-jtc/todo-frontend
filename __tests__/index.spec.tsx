import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import HomePage from '../pages/index';
import {
  ButtonInput,
  InputField,
  ImageLogo,
  AppComponent,
  Header,
} from '@components';

jest.mock('next/image', () => ({
  // eslint-disable-next-line @typescript-eslint/naming-convention
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line jsx-a11y/alt-text
    return <img {...props} />;
  },
}));

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

jest.mock('next-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('With Snapshot Testing', () => {
  it('App shows Hello World heading and form', () => {
    const component = renderer.create(<HomePage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('With Snapshot Testing', () => {
  it('App shows Button', () => {
    const component = renderer.create(
      <ButtonInput onClick={() => {}} text='login' isDisabled={false} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('With Snapshot Testing', () => {
  it('App shows Input Field', () => {
    const component = renderer.create(
      <InputField
        name='email'
        type='email'
        label='Email'
        onChange={() => {}}
        onBlur={() => {}}
        value=''
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('With Snapshot Testing', () => {
  it('App shows Image', () => {
    const component = renderer.create(<ImageLogo alt='' src='' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('With Snapshot Testing', () => {
  it('App shows App Component', () => {
    const component = renderer.create(
      <AppComponent />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('With Snapshot Testing', () => {
  it('App shows header', () => {
    const component = renderer.create(
      <Header text_btn1='Engish' text_btn2='Francis' />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
