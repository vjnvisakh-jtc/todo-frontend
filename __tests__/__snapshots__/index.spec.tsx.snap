// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows App Component 1`] = `
Array [
  <div
    className="head"
  >
    <p
      className="heading"
    >
      title
    </p>
    <p
      className="sub-head"
    >
      about_para
       
      <span>
        <a
          href="https://github.com/jalantechnologies/boilerplate-next-typescript#readme"
          style={
            Object {
              "color": "#7D7D7D",
            }
          }
        >
          readme
        </a>
      </span>
       
    </p>
  </div>,
  <div
    className="next-logo"
  />,
  <div
    className="base-logo"
  />,
  <div
    className="eslint-logo"
  />,
  <div
    className="storybook-logo"
  />,
  <div
    className="seo-logo"
  />,
  <div
    className="ts-logo"
  />,
  <div
    className="sentry-logo"
  />,
  <div
    className="i18n-logo"
  />,
  <div
    className="jest-logo"
  />,
  <div
    className="formik-logo"
  />,
]
`;

exports[`With Snapshot Testing App shows Button 1`] = `
<button
  className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
  data-baseweb="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  login
</button>
`;

exports[`With Snapshot Testing App shows Hello World heading and form 1`] = `
Array [
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />,
  <link
    href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap"
    rel="stylesheet"
  />,
  <link
    href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta3/dist/css/bootstrap.min.css"
    integrity="sha384-eOJMYsd53ii+scO/bJGFsiCZc+5NDVN2yr8+0RDqr0Ql0h+rP48ckxlpbzKgwra6"
    rel="stylesheet"
  />,
  <header>
    <div
      className="btn-en"
    >
      <button
        className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
        data-baseweb="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        btn-text1
      </button>
    </div>
    <div
      className="btn-fr"
    >
      <button
        className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
        data-baseweb="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        btn-text2
      </button>
    </div>
  </header>,
  <div>
    <main>
      <div
        className="head"
      >
        <p
          className="heading"
        >
          title
        </p>
        <p
          className="sub-head"
        >
          about_para
           
          <span>
            <a
              href="https://github.com/jalantechnologies/boilerplate-next-typescript#readme"
              style={
                Object {
                  "color": "#7D7D7D",
                }
              }
            >
              readme
            </a>
          </span>
           
        </p>
      </div>
      <div
        className="next-logo"
      />
      <div
        className="base-logo"
      />
      <div
        className="eslint-logo"
      />
      <div
        className="storybook-logo"
      />
      <div
        className="seo-logo"
      />
      <div
        className="ts-logo"
      />
      <div
        className="sentry-logo"
      />
      <div
        className="i18n-logo"
      />
      <div
        className="jest-logo"
      />
      <div
        className="formik-logo"
      />
    </main>
  </div>,
]
`;

exports[`With Snapshot Testing App shows Image 1`] = `
<img
  alt=""
  height={200}
  src=""
  width={200}
/>
`;

exports[`With Snapshot Testing App shows Input Field 1`] = `
<div
  className="input-wrapper"
>
  <label
    htmlFor="email"
  >
    Email
  </label>
  <br />
  <div
    className="bl bm bn bo bp bq br bs bt bu bv bw bx av by bz c0 c1 c2 b5 b6 c3 c4 c5 c6 c7 c8 bi c9 ca"
    data-baseweb="input"
  >
    <div
      className="bm bo cb av by b5 b6 c3 c4 bh bi"
      data-baseweb="base-input"
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete={null}
        aria-controls={null}
        aria-describedby={null}
        aria-errormessage={null}
        aria-invalid={false}
        aria-label={null}
        aria-labelledby={null}
        aria-required={false}
        autoComplete="on"
        className="bl cc ai ak aj al am ao an ap aq bo cd ce cf cg ch ci cj ck b5 b6 c3 c4 bh cl cm"
        disabled={false}
        id="email"
        inputMode="text"
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        pattern={null}
        placeholder="Email"
        required={false}
        rows={null}
        type="email"
        value=""
      />
    </div>
  </div>
  <br />
</div>
`;

exports[`With Snapshot Testing App shows header 1`] = `
<header>
  <div
    className="btn-en"
  >
    <button
      className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
      data-baseweb="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Engish
    </button>
  </div>
  <div
    className="btn-fr"
  >
    <button
      className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk"
      data-baseweb="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Francis
    </button>
  </div>
</header>
`;
